zaza/openstack/charm_tests/audit/tests.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+#
+# Copyright 2024 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Keystone audit middleware API logging testing.
+
+These methods test the
+rendering of the charm's api-paste.ini file to ensure the appropriate sections
+are rendered or not rendered depending on the state of the audit-middleware
+configuration option.
+"""
+
+import logging
+import zaza.model
+import zaza.openstack.charm_tests.test_utils as test_utils
+
+
+class KeystoneAuditMiddlewareBaseTest(test_utils.OpenStackBaseTest):
+    """Base class for Keystone audit middleware tests."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Run class setup for Keystone audit middleware tests."""
+        super(KeystoneAuditMiddlewareBaseTest, cls).setUpClass()
+        test_config = cls.test_config['tests_options']['audit-middleware']
+        cls.service_name = test_config['service']
+
+        # For cases when service name and application name differ
+        if not test_config["application"]:
+            cls.application_name = cls.service_name
+        else:
+            cls.application_name = test_config['application']
+        print(test_config)
+        print(cls.model_name)
+        cls.lead_unit = zaza.model.get_lead_unit_name(
```

Comment:
this could change over time, storing the leader unit during the `setUpClass` may lead to inaccurate results, it's better you get the lead unit closer to where it will be used (e.g. the test method).